admin-auth/src/test/kotlin/com/epam/drill/admin/auth/CacheServiceImplTest.kt

```diff
@@ -0,0 +1,48 @@
+package com.epam.drill.admin.auth
+
+import com.epam.drill.admin.auth.principal.Role
+import com.epam.drill.admin.auth.principal.User
+import com.epam.drill.admin.auth.service.impl.CacheServiceImpl
+import com.github.benmanes.caffeine.cache.Caffeine
+import kotlinx.coroutines.runBlocking
+import org.junit.Assert
+import java.time.Duration
+import kotlin.test.Test
+
+/**
+ * Tests for [CacheServiceImpl] logic
+ */
+class CacheServiceImplTest {
+
+    @Test
+    fun `given two distinct apiKeys, then cache size should be two`(): Unit = runBlocking {
+        val cacheService = CacheServiceImpl(
+            Caffeine.newBuilder()
+                .maximumSize(5)
+                .expireAfterWrite(Duration.ofMinutes(1))
+                .build()
+        )
+
+        cacheService.getFromCacheOrPutIfAbsent("custom-key-1") { User(1, "test1", Role.USER) }
+        cacheService.getFromCacheOrPutIfAbsent("custom-key-2") { User(2, "test2", Role.USER) }
+
+        Assert.assertEquals(2, cacheService.cache.estimatedSize())
+    }
+
+    @Test
+    fun `given only one apiKey two times, then cache size should be one`(): Unit = runBlocking {
```

Comment:
I'm not as good at writing tests as @RomanDavlyatshin , but after going through a lot of reviews from him, I can give some advice on how to improve this test.

The purpose of testing is not to control the cache size, but the fact that we do not access the API key service if the result is already in the cache. 
Regarding this purpose will be better rename test case like 
`if api key is already in cache, getFromCacheOrPutIfAbsent function must return user from cache`.
And we will have to rework the test code so that the `getFromCacheOrPutIfAbsent` function is called only once.
Like this:
```
var testApiKey = "test-api-key"
val testUser = ...
val cache = Coffeine. ... .build()
  .also { put(testApiKey, testUser) }

val user = cacheService.getFromCacheOrPutIfAbsent(testApiKey) { assertFail("Must not be called") }

assertEquals(testUser, user)
```

Reply:
Refactored. The changes are open for discussion and review.